### PR TITLE
[4.4] Fix next exec calculation for task scheduler

### PR DESCRIPTION
**Pull request by rdeutz**
Pull Request for Issue #42459 partly .

### Summary of Changes
The calculation of the next execution is not working as expected. This fix is for the intervals:
* interval-minutes
* interval-hours
* interval-days
* interval-months

I haven't looked into the cron-execution but will do in the furture (seems to me working right).

### Testing Instructions
Create a task (e.g. Sleep) and chose Interval Days and set "Interval in Days" to 1 and the "Execution Time (UTC)" to 11:11 and save it.

### Actual result BEFORE applying this Pull Request
You will see that the day is the next day but the time is the current time. 

### Expected result AFTER applying this Pull Request
The day is the next day and the time is 11:11

Apply the patch and try again, now it should be as expected. Play around with the other options and settings.

For the calculation of the next execution I have used the current time, then I am adding the interval and finally set time and day if needed.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: administrator/components/com_scheduler/src/Helper/ExecRuleHelper.php [hunk 1]
```diff
@@ -93,9 +93,50 @@ private function getFromTask(string $property, $default = null)
...
+                $nextExecMonth  = $nextExec->format('n');
+                $nextExec->setDate($nextExecYear, $nextExecMonth, $execDay);
+
+                $execTime            = $executionRules['exec-time'];
+                list($hour, $minute) = explode(':', $execTime);
+                $nextExec->setTime($hour, $minute);
+                $nextExec = $string ? $nextExec->toSql() : $nextExec;
+                break;
             case 'interval':
+                // We keep this for b/c reasons, can be removed in 7.0.0
```
Comment (sandewt) [inline administrator/components/com_scheduler/src/Helper/ExecRuleHelper.php]:
```suggestion
                // @deprecated 4.4 will be removed in 7.0
```
Is there a reason to drop this in 7.0.0, and not in 6.0?

Comment (rdeutz) [reply administrator/components/com_scheduler/src/Helper/ExecRuleHelper.php]: We have the same error in 5.x, so I used 7 instead of 6.

**Comment by TLWebdesign** [comment]
I have tested this item :white_check_mark: successfully on 52f7cb73460ec9792e9c5cd77e87612c5d00f721<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44061">issues.joomla.org/tracker/joomla-cms/44061</a>.</sub>

**Comment by dautrich** [comment]
I have tested this item :white_check_mark: successfully on 52f7cb73460ec9792e9c5cd77e87612c5d00f721<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44061">issues.joomla.org/tracker/joomla-cms/44061</a>.</sub>

**Comment by Quy** [comment]
Not sure to mark RTC per [comment](https://github.com/joomla/joomla-cms/issues/42459#issuecomment-2350674336).

**Comment by rdeutz** [comment]
@Quy I would set RTC because it is a good step in the right direction and we only have a few days before the last 4.4.x Version before securtiy only fixes will be released. It would be a shame to have it so broken as it is now. I am going to make a 2nd PR and will work on the other issues in the comment.

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44061">issues.joomla.org/tracker/joomla-cms/44061</a>.</sub>

**Comment by MacJoom** [comment]
Thank you!

**Pull request merged**